src/common/darktable.c

```diff
@@ -160,13 +164,13 @@ gboolean dt_supported_image(const gchar *filename)
 static void strip_semicolons_from_keymap(const char *path)
 {
   char pathtmp[PATH_MAX] = { 0 };
-  FILE *fin = fopen(path, "r");
+  FILE *fin = fopen(path, "rb");
```

Comment:
I am not convinced that `rb` is a good idea. That will work with the files shipped with darktable, but once the user edits them with a Windows text editor he will have Windows line endings in the file and it will break again. Instead you could add checks for `\r` next to then `\n` checks and properly skip them.